### PR TITLE
n_queens.py added under backtracking_algorithms issure #56

### LINKED ISSUE #56: Add Backtracking Algorithms
These are the popular algorithms, that use backtracking. Your task is to design the code for these algorithms inside the package.

- Print all permutations of a given string(@Prakhar-Mangal)
- The Knight’s tour problem (@Priyaa-Agrawal)
- Rat in a Maze (@charlie219)
- N Queen Problem (@charlie219 )
- Subset Sum (@vedantgoswami)
- m Coloring Problem(@Prakhar-Mangal)
- Hamiltonian Cycle
- Sudoku (@vedantgoswami)
- Tug of War
- Solving Cryptarithmetic Puzzles

As these are quite popular, you just need to google the name for the problem statement, but never use google for the solution, try to solve it first, if not possible, try again. Try two times more.

As a final step, you may contact the project admin. But never use google for a solution directly.

### PR DESCRIPTION
**Pull request by charlie219**

### CODE REVIEW

File: edualgo/backtracking_algorithms/n_queen.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
+# N-Queens Problem via Backtracking
+
+from __init__ import print_msg_box
+from time import time as ti
+n=0                                             #Size of the Board
```
Comment (Abhijit2505) [inline edualgo/backtracking_algorithms/n_queen.py]: There is no need to take input, those will be passed by the user manualy as a parameter of the function.

File: edualgo/backtracking_algorithms/n_queen.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+ans=[]                                          #Answer in the form of list of co-ordinates to place the Queens
+
+#Occupied Diagonals and Columns
+diagonal1={};diagonal2={}                       #For right and left Diagonal respectively
+Col={}                                          #For Column which are already alloted to some queen 
+
+def n_queens(board_size,hint=False):            #Main function argument =size of the board
```
Comment (Abhijit2505) [inline edualgo/backtracking_algorithms/n_queen.py:6]: Place the main code inside a function such that any user can call the function and get the work done, for more details refer to the codebase and project documentations.

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]

File: edualgo/backtracking_algorithms/n_queen.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    #Occupied Diagonals and Columns
+    diagonal1={};diagonal2={}                       #For right and left Diagonal respectively
+    Col={}                                          #For Column which are already alloted to some queen 
+    ans=place_queen(0,[],board_size,diagonal1,diagonal2,Col)                      
+    print(ti()-st)
+
+    if not ans:
+        return -1
+
+    if hint:
```
Comment (Abhijit2505) [inline edualgo/backtracking_algorithms/n_queen.py]: This method should be above line 16, isn't it ?

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]

**Comment by charlie219** [comment]
global variables removed as requested @Abhijit2505

**Comment by Abhijit2505** [comment]
> global variables removed as requested @Abhijit2505

There is already a minor change requested above.

File: edualgo/backtracking_algorithms/n_queen.py (hunk 3 above)
Comment (charlie219) [reply edualgo/backtracking_algorithms/n_queen.py]: yes, got your point.
Anything more to be changed?

File: edualgo/backtracking_algorithms/n_queen.py (hunk 2 above)
Comment (charlie219) [reply edualgo/backtracking_algorithms/n_queen.py:6]: the user can just call the function n_queens() with the size of the board as the argument and the work will be done

**Comment by charlie219** [comment]
@Abhijit2505, please review the code and tell me if I m missing something

**Review by Siddhant-K-code — APPROVED** [review]

**Review by Abhijit2505 — APPROVED** [review]

**Pull request merged**